### PR TITLE
Use ClimaComms environment variable only in pipeline

### PR DESCRIPTION
**Pull request by charleskawczynski**
This PR switches over to using the latest ClimaComms environment variable.

### CODE REVIEW

File: test/Spaces/distributed_cuda/ddss2.jl [hunk 1]
```diff
@@ -7,7 +7,7 @@ import ClimaCore:
 using ClimaComms
 using CUDA
 
-ENV["CLIMACORE_DISTRIBUTED"] = "MPI"
+ENV["CLIMACOMMS_CONTEXT"] = "MPI"
 
 # initializing MPI
 const device = ClimaComms.device()
```
Comment (simonbyrne) [inline test/Spaces/distributed_cuda/ddss2.jl:13]:
```suggestion
const device = ClimaComms.MPICommsContext()
```

File: test/Spaces/distributed/gather4.jl [hunk 2]
```diff
@@ -13,7 +13,7 @@ import ClimaCore:
 
 using Logging
 
-ENV["CLIMACORE_DISTRIBUTED"] = "MPI"
+ENV["CLIMACOMMS_CONTEXT"] = "MPI"
```
[the remark this answers is not in the document]
Comment (simonbyrne) [reply test/Spaces/distributed/gather4.jl:16]: Oh i was wrong. I meant `context`, which has already been set

**Comment by simonbyrne** [comment]
So I think the variable is redundant, and can be removed altogether if `MPICommsContext` is already used.

File: test/Spaces/distributed/gather4.jl (hunk 2 above)
[the remark this answers is not in the document]
Comment (charleskawczynski) [reply test/Spaces/distributed/gather4.jl:16]: Right, that's what I thought you meant

**Comment by charleskawczynski** [comment]
> So I think the variable is redundant, and can be removed altogether if `MPICommsContext` is already used.

Right, I think the only time using the environment variable is useful is if we use `ClimaComms.context()`, which then relies on the environment variable, correct?

**Comment by simonbyrne** [comment]
Yes exactly

**Comment by charleskawczynski** [comment]
Alright, @simonbyrne, I was able to remove the use of `CLIMACOMMS_CONTEXT` from src with that suggestion.

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

**Comment by simonbyrne** [comment]
That's unrelated: the build that is failing is timing out

**Comment by charleskawczynski** [comment]
Should be good to go now.

File: examples/hybrid/driver.jl [hunk 3]
```diff
@@ -23,16 +23,14 @@ face_initial_condition(local_geometry) = (;)
 postprocessing(sol, output_dir) = nothing
 
 ################################################################################
-is_distributed = get(ENV, "CLIMACOMMS_CONTEXT", "") == "MPI"
 
-using Logging
 using ClimaComms
+const comms_ctx = ClimaComms.context()
```
Comment (simonbyrne) [inline examples/hybrid/driver.jl:28]: `comms_ctx` is still redefined again in the other branch to `is_distributed`

**Pull request merged**